src/library_tty.js

```diff
@@ -63,13 +63,24 @@ mergeInto(LibraryManager.library, {
           if (result === null || result === undefined) break;
           bytesRead++;
           buffer[offset+i] = result;
+
+          // If EOL character, return having put it in the buffer.
+          if (result === 10) break;
```

Comment:
Hmm, why this?

Reply:
Currently 'read' doesn't return until you hit CTRL-D to send EOF (or fill the buffer).  This is not obvious from the automated tests, but clearly isn't what you want from an interactive session, where it should return when you press RETURN.

Reply:
Thanks, this I think is something we should have an interactive test for, it is quite a specific scenario.